### PR TITLE
Push bytecode register allocation to JIT compile time

### PR DESCRIPTION
**Pull request by guillep**
Extend register allocation to decide actual registers are JIT compile time and not ahead of time

### CODE REVIEW

File: Druid-Tests/DRCogitJITConstantSplittingTest.class.st [hunk 1]
```diff
@@ -1,62 +0,0 @@
-Class {
-	#name : #DRCogitJITConstantSplittingTest,
-	#superclass : #TestCase,
-	#category : #'Druid-Tests'
-}
```
Comment (PalumboN) [inline Druid-Tests/DRCogitJITConstantSplittingTest.class.st:5]: Why these tests were removed? 🤔

File: Druid-Tests/DruidTestRTLCompiler.class.st [hunk 2]
```diff
@@ -3254,10 +3254,11 @@ DruidTestRTLCompiler >> gen_pushLiteralVariable16CasesBytecode [
 DruidTestRTLCompiler >> gen_pushReceiverBytecode [
 	"AutoGenerated by Druid"
 
-	| currentBlock |
+	| R6 currentBlock |1 halt.
```
Comment (PalumboN) [inline Druid-Tests/DruidTestRTLCompiler.class.st:3257]: Halt!

Anyway, this method is generated.

File: Druid/DRCogitJITConstantSplitting.class.st [hunk 3]
```diff
@@ -1,35 +0,0 @@
...
-	A := 1 + 2
-	=>
-	B := 1
-	A := B + 2
-"
-Class {
-	#name : #DRCogitJITConstantSplitting,
-	#superclass : #DROptimisation,
-	#category : #'Druid-Cogit'
-}
```
Comment (PalumboN) [inline Druid/DRCogitJITConstantSplitting.class.st:15]: Ahh ok... this is not necessary anymore 👍

File: Druid/DRInfiniteRegisterPool.class.st [hunk 4]
```diff
@@ -0,0 +1,5 @@
+Class {
+	#name : #DRInfiniteRegisterPool,
+	#superclass : #Object,
+	#category : #'Druid-BytecodeToJITCompilation'
+}
```
Comment (PalumboN) [inline Druid/DRInfiniteRegisterPool.class.st:5]: 😱 ♾️ 

Is it used anywhere?

**Review by PalumboN — APPROVED** [review]
Cool! 👍 

Take care, there are some tests failing. Including `testBytecodePushLiteralVariable`, it's ok?

File: Druid-Tests/DRProductionBytecodeTest.class.st [hunk 5]
```diff
@@ -302,6 +302,84 @@ DRProductionBytecodeTest >> testBytecodePushLiteralVariable: tempIndex [
...
+	tempIndex := 0. "0 based..."
+	self
+		compileBytecode: 64 + tempIndex
+		selector: #pushTemporaryVariableBytecode
+		thenDo: [ :generator | "Execute the druid's compiled code"
+			cogit methodOrBlockNumArgs: 1.
+			cogit methodOrBlockNumTemps: 1.
+			cogit initSimStackForFramefulMethod: 2.
+
+			cogit gen_pushReceiverBytecode.
```
[the remark this answers is not in the document]
Comment (guillep) [reply Druid-Tests/DRProductionBytecodeTest.class.st:364]: I'll check

File: Druid-Tests/DruidTestRTLCompiler.class.st (hunk 2 above)
Comment (guillep) [reply Druid-Tests/DruidTestRTLCompiler.class.st:3257]: Yes, I'll check, running the test should override it

File: Druid/DRCogitJITConstantSplitting.class.st (hunk 3 above)
Comment (guillep) [reply Druid/DRCogitJITConstantSplitting.class.st:15]: Yup

File: Druid/DRInfiniteRegisterPool.class.st (hunk 4 above)
Comment (guillep) [reply Druid/DRInfiniteRegisterPool.class.st:5]: Ah, no, a leftover!

File: Druid-Tests/DruidTestRTLCompiler.class.st (hunk 2 above)
Comment (guillep) [reply Druid-Tests/DruidTestRTLCompiler.class.st:3257]: Fix coming

**Pull request merged**